src/main/java/eu/darkbot/api/managers/LogAPI.java

```diff
@@ -2,19 +2,31 @@
 
 import eu.darkbot.api.API;
 import eu.darkbot.api.objects.Gui;
+import eu.darkbot.api.utils.Listener;
 
 import java.util.Collection;
+import java.util.Optional;
 
 /**
  * API for in-game log messages
  */
 public interface LogAPI extends Gui, API {
+
+    /**
+     * Adds {@link Listener} which will be called on each new message.
+     *
+     * @param onMessage {@link Listener} to be added
+     * @return reference to {@code onMessage} listener
+     * @see Listener
+     */
+    Listener<String> addListener(Listener<String> onMessage);
+
     /**
      * Gets the most recent log message
      *
      * @return the most recent message in your in-game log
      */
-    String getLastMessage();
+    Optional<String> getLastMessage();
```

Comment:
I think this method shouldn't even exist, it may confuse developers thinking they can stay up-to-date with log with this, but you'll get 4 log messages at once and miss 3 of them